### ISSUE TITLE
Well-integrated support for ClutTeX compiler

### ISSUE THREAD

**Issue by paniash** [lervag/vimtex#2116]
**Is your feature request related to a problem? Please describe it.**

It would be a nice addition to add integrated support for ClutTeX instead of relying on the generic compiler usage.

**Describe the solution you'd like**

Similar to support for latexmk, ClutTeX could be added as an option for specifying the compiler backend.

ClutTeX is primarily written in lua and now that neovim supports it, I think it would be well integrated. A major selling point of ClutTeX is that is does everything latexmk does (AFAIK) and doesn't clutter the current directory (which as opposed to latexmk has to be cleaned up via `latexmk -c`).

**Comment by lervag** [comment]
I don't think it should be too hard, but it will take some time and I consider it a low priority request. I will not mind a PR, if anyone should have time and interest to work on it.

**Comment by jliptrap** [comment]
latexmk doesn't have to clutter the current directory, here's a line of vimtex config to put all that clutter in a `build` subdir:

```vim
let g:vimtex_compiler_latexmk = {'build_dir' : 'build'}
```

**Comment by GersiD** [comment]
Equivalent Lua for anyone wondering 
```
        vim.g.vimtex_compiler_latexmk = { build_dir = "build" }
```

**Comment by lervag** [comment]
I'll just have to admit that I'll probably never really have the time for this. I'm closing as unplanned, but I will accept PRs if anyone else wants to build it.
